test/libsolidity/semanticTests/structs/conversion/recursive_storage_memory.sol

```diff
@@ -1,20 +1,26 @@
 contract CopyTest {
-	struct Tree {
-		Tree[] children;
-	}
-	Tree storageTree;
+    struct Tree {
+        Tree[] children;
+    }
+    Tree storageTree;
+    Tree childStorageTree;
 
-	constructor() public {
-		storageTree.children.length = 2;
-		storageTree.children[0].children.length = 23;
-		storageTree.children[1].children.length = 42;
-	}
+    Tree[] children;
 
-	function run() public returns (uint256, uint256, uint256) {
-		Tree memory memoryTree;
-		memoryTree = storageTree;
-		return (memoryTree.children.length, memoryTree.children[0].children.length, memoryTree.children[1].children.length);
-	}
+    constructor() public {
+        for (uint i = 0; i < 2; i++)
+            storageTree.children.push(childStorageTree);
+        for (uint i = 0; i < 23; i++)
+            storageTree.children[0].children.push(childStorageTree);
```

Comment:
Won't this push the same reference all over?

Reply:
Ah, true. This is a left over and I now changed it to empty `push()`.